### PR TITLE
Add initial configuration for fleet-server

### PR DESCRIPTION
**Pull request by blakerouse**
## Overview

This adds the initial configuration for the Fleet Server.

Adds 3 sections to the configuration:

- `elasticsearch`: configuration for communication with Elasticsearch (requires username/password, API key errors)
- `server`: configuration for the HTTP server that fleet-server runs for the Elastic Agents to connect to (missing server-side TLS options)
- `logging` configuration for the logging

At the moment only `logging` is actually used, the others need to be plumbed in.

Adds `-c` and `--config` to specify the configuration of Fleet Server as well as `--help` using `cobra`.

Adds default configuration that pulls settings from ENV so running it in container can be configured. But this really should not be needed as it should only really be executable from Elastic Agent, where it passes those settings down.

### CODE REVIEW

**Review by scunningham — APPROVED** [review]
👍   Thanks for keeping the logger in place.

File: internal/pkg/config/server.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// InitDefaults initializes the defaults for the configuration.
+func (c *ServerTimeouts) InitDefaults() {
+	c.Read = 5        // 5 seconds
+	c.Write = 60 * 10 // 10 minutes (long poll)
+}
+
+// Server is the configuration for the server
+type Server struct {
+	Host     string         `config:"host"`
+	Port     uint16         `config:"host"`
```
Comment (ruflin) [inline internal/pkg/config/server.go:35]: `config:"host"` should be `config:"port"`

File: internal/pkg/config/server.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package config
+
+// ServerTimeouts is the configuration for the server timeouts
+type ServerTimeouts struct {
```
Comment (ruflin) [inline internal/pkg/config/server.go:21]: Any chance to add these also to the fleet-server.yml file so we have all the configs in a single place "documented"?

File: internal/pkg/config/logging.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/rs/zerolog"
+)
+
+// Logging is the logging configuration
+type Logging struct {
```
Comment (ruflin) [inline internal/pkg/config/logging.go:30]: Same for these config options, please add them to the fleet-server config file.

**Pull request merged**